src/error_handler.py

```diff
@@ -1,13 +1,17 @@
-import os
+from __future__ import annotations
+
 import sys
 from enum import Enum
 
+
 def eprint(s):
     print(s, file=sys.stderr)
 
+
 def abort(s):
     eprint(s)
-    os._exit(1)
+    sys.exit(1)
```

Comment:
Hmm.  I actually was using `os._exit` here intentionally to avoid having the exception bubbling out.  The `sys.exit` call apparently causes a SystemExit exception to bubble up to the higher level handlers.  In some cases, thought, I just want the program to immediately exit.

https://stackoverflow.com/questions/173278/is-there-a-way-to-prevent-a-systemexit-exception-raised-from-sys-exit-from-bei

Reply:
I think we might want to look into this a bit more for a less heavy-handed way of accomplishing the same thing. `os._exit()` is a sledgehammer.

Reply:
I'm going to punt on this for this change and revert back to os._exit().

I gotta say I'm still shocked by the number of foot guns in this language.

Reply:
That sounds like a good plan.  I'll take the responsibility to go back and fix it later.